### PR TITLE
crystal plugin: add flags to use during shards build

### PR DESCRIPTION
**Pull request by mamantoha**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/plugins/crystal.py [hunk 1]
```diff
@@ -80,7 +94,12 @@ def build(self):
         super().build()
 
         self.run(["shards", "install", "--production"], self.builddir)
-        self.run(["shards", "build", "--production"], self.builddir)
+
+        build_flags = ["--production"]
```
Comment (sergiusens) [inline snapcraft/plugins/crystal.py]: we should probably let the defaults take place here, how would I build a non production build if I wanted to?

Comment (mamantoha) [reply snapcraft/plugins/crystal.py]: Do you mean here?

```
schema["properties"]["crystal-buildflags"] = {
            "type": "array",
            "minitems": 1,
            "uniqueItems": True,
            "items": {"type": "string"},
            "default": [], # << HERE
        }
```

**Comment by bcardiff** [comment]
Regarding the crystal management of options, it looks good to me.

Thanks @mamantoha !

File: snapcraft/plugins/crystal.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/plugins/crystal.py]: That is a good place, please make mention of the default in the docstring.

Comment (mamantoha) [reply snapcraft/plugins/crystal.py]: Done

Comment (sergiusens) [reply snapcraft/plugins/crystal.py]: You seem to have missed adding it to "default" in the schema

Comment (mamantoha) [reply snapcraft/plugins/crystal.py]: Done

Comment (sergiusens) [reply snapcraft/plugins/crystal.py]: ok, so now this line is not needed as it should be taken into account in the conditional right below

**Pull request merged**